pkg/communication/transport_endpoint.go

```diff
@@ -0,0 +1,43 @@
+package communication
+
+import "fmt"
+
+// Transport endpoint that sends and receives raw message bytes
+type ITransport interface {
+	CloseTransportPoint()
+	Send(messageToSend *TransportMessage)
+	RawMessageReceiver() chan []byte
+}
+
+type TransportEventHandler interface {
+	EventReceiver() chan []byte
+	OnClose()
+	OnMessage(msgReceived []byte)
+}
+
+type TransportMessage struct {
```

Comment:
If TransportMessage is only a slice of byte, we can use
```
type TransportMessage []byte
```
instead of defining a separate struct.